### PR TITLE
RES&COMP: resolve and complete paths in include macros

### LINKED ISSUE #1952: Autocompletion for include_bytes & include_str macros
`include_bytes!` and `include_str!` macros accept path as an argument. When you type the path IDE can show autocompletion with the list of files available on the filesystem.

### LINKED ISSUE #2533: "Go to Declaration" should work on include! statements
Right now, if I have a line like this:

```rust
include!("some_file.rs");
```

"Go to Declaration" over that filename does not do anything. It would be nice if it opened the target file.

### PR DESCRIPTION
**Pull request by Undin**
Closes #1952
Closes #2533

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+
+private class RsPathReferenceProvider : PsiReferenceProvider() {
+
+    override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<FileReference> {
+        val stringLiteral = (element as? RsLitExpr)?.kind as? RsLiteralKind.String ?: return emptyArray()
+        if (stringLiteral.isByte) return emptyArray()
+        val fs = element.containingFile.originalFile.virtualFile.fileSystem
+        val startOffset = ElementManipulators.getValueTextRange(element).startOffset
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt]: We can replace it with more simple `stringLiteral.offsets.value.startOffset`

File: src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.rust.lang.core.psi.kind
+
+class RsLitExprReferenceContributor : PsiReferenceContributor() {
+
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        registrar.registerReferenceProvider(RsPsiPattern.includeMacroLiteral, RsPathReferenceProvider())
+    }
+}
+
+private class RsPathReferenceProvider : PsiReferenceProvider() {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt]: Maybe rename it to "RsFileReferenceProvider"?

**Comment by Undin** [comment]
bors retry

**Comment by Undin** [comment]
bors r=vlad20012

**Comment by Undin** [comment]
bors r=vlad20012

**Comment by Undin** [comment]
bors r=vlad20012

**Comment by ortem** [comment]
bors retry

**Comment by Undin** [comment]
bors try

**Comment by Undin** [comment]
bors r=vlad20012

**Pull request merged**